frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -586,41 +606,99 @@ export default class FirebaseInteractor {
     return totalWords;
   }
 
-  async createDataZip(userId: string, name: string) {
+  async createAllDataZip() {
+    const zip = new JSZip();
+    const folder = zip.folder("student_data");
+    if (folder === null) {
+      throw new Error("Error downloading data. Please try again.");
+    }
+    const students = await this.getAllStudents();
+    const promises = students.map((student) =>
+      this.getDataForOneUser(student.id, student.name, folder).catch((error) =>
+        console.log(error)
+      )
+    );
+    await Promise.all(promises);
+    let fileBlob = await zip.generateAsync({ type: "blob" });
```

Comment:
why is it called a blob lol (not an important question)

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Blob